### PR TITLE
perf: cleanup atomics

### PR DESCRIPTION
**Pull request by h-a-n-a**
## Require Documentation?

- [ ] No
- [ ] Yes, the corresponding rspack-website PR is \_\_

### CODE REVIEW

File: crates/rspack_core/src/cache/mod.rs [hunk 1]
```diff
@@ -53,16 +53,12 @@ impl Cache {
   }
 
   pub fn begin_idle(&self) {
-    if self
-      .is_idle
-      .compare_exchange(false, true, Ordering::AcqRel, Ordering::Relaxed)
-      .is_ok()
-    {
+    if self.is_idle.load(Ordering::Relaxed) {
```
Comment (h-a-n-a) [inline crates/rspack_core/src/cache/mod.rs:56]: Happens on the same thread, not necessary for `compare_exchange` and `AcqRel`

File: crates/rspack_core/src/compiler/make/mod.rs [hunk 2]
```diff
@@ -233,12 +233,12 @@ impl UpdateModuleGraph {
           let is_expected_shutdown = self.is_expected_shutdown.clone();
 
           async move {
-            if is_expected_shutdown.load(Ordering::SeqCst) {
+            if is_expected_shutdown.load(Ordering::Relaxed) {
```
Comment (h-a-n-a) [inline crates/rspack_core/src/compiler/make/mod.rs:236]: Atomicity is enough, no other synchronization guarantees are necessry.

File: crates/rspack_plugin_progress/src/lib.rs [hunk 3]
```diff
@@ -66,11 +66,11 @@ impl ProgressPlugin {
   }
 
   fn update(&self) {
-    let modules_done = self.modules_done.load(SeqCst);
+    let modules_done = self.modules_done.load(Relaxed);
```
Comment (h-a-n-a) [inline crates/rspack_plugin_progress/src/lib.rs:69]: Same here.

File: crates/rspack_tracing/src/lib.rs [hunk 4]
```diff
@@ -30,7 +31,7 @@ impl<S> Filter<S> for FilterEvent {
 }
 
 pub fn enable_tracing_by_env(filter: &str, output: &str) {
-  if !IS_TRACING_ENABLED.swap(true, std::sync::atomic::Ordering::SeqCst) {
+  if !IS_TRACING_ENABLED.swap(true, Ordering::Relaxed) {
```
Comment (h-a-n-a) [inline crates/rspack_tracing/src/lib.rs:34]: Same thread.

**Review by ahabhgk — APPROVED** [review]

**Pull request merged**